mantidproject/mantid
# Add Bk2BkExpConvPV parameters for ENGINX and improve FWHM of function

**Description of work.**

- Add parameters for `Bk2BkExpConvPV` for ENGINX.
- Fix bug in pseud-voigt FWHM, `H`, and the mixing parameter `eta` - it should use the FWHM of Lorz (`2*gamma`) [1]
- Add setter for FWHM using Eq. in `BackToBackExponential` (which only has Gaussian component). Assuming the Voigt has same FWHM as Gaussian in `BackToBAckExponential` then solving Eq. for `H` assuming `H_L = H_G` (i.e. Gaussian and Lorz components of voigt have same FWHM).

[1]  P. Thompson, D. E. Cox and J. B. Hastings (1987). "Rietveld refinement of Debye-Scherrer synchrotron X-ray data from Al2O3". Journal of Applied Crystallography. 20 (2): 79–83. doi:10.1107/S0021889887087090.

**To test:**

(1) Run this script to make some data 
```
Load(Filename=r'\\isis.cclrc.ac.uk\inst$\ndxenginx\instrument\data\cycle_12_4\ENGINX00193749.nxs', OutputWorkspace='193749')
ConvertUnits(InputWorkspace='193749', OutputWorkspace='193749', Target='dSpacing')
CreateGroupingWorkspace(InputWorkspace='193749', GroupNames='NorthBank', 
    OutputWorkspace='grouping')
DiffractionFocussing(InputWorkspace='193749', OutputWorkspace='foc', 
    GroupingWorkspace='grouping')
ConvertUnits(InputWorkspace='foc', OutputWorkspace='foc', Target='TOF')

foc = ADS.retrieve('foc')
func = Bk2BkExpConvPV()
func.setParameter(0, 35219.7) # X0
func.setMatrixWorkspace(foc, 0, 0.0,0.0) # calculate A,B,S
print([func.getParamValue(ipar) for ipar in range(func.function.nParams())] )
```
It should print
`[35219.7, 0.0, 0.143624, 0.0207591, 1250.95, 9.85146]`

(2) Make this plot
```
fig, axes = plt.subplots(edgecolor='#ffffff', num='foc-1', subplot_kw={'projection': 'mantid'})
axes.plot(foc, color='#1f77b4', label='foc: spec 1', marker='.', wkspIndex=0)
axes.tick_params(axis='x', which='major', **{'gridOn': False, 'tick1On': True, 'tick2On': False, 'label1On': True, 'label2On': False, 'size': 6, 'tickdir': 'out', 'width': 1})
axes.tick_params(axis='y', which='major', **{'gridOn': False, 'tick1On': True, 'tick2On': False, 'label1On': True, 'label2On': False, 'size': 6, 'tickdir': 'out', 'width': 1})
axes.set_title('foc')
axes.set_xlabel('Time-of-flight ($\\mu s$)')
axes.set_ylabel('Counts ($\\mu s$)$^{-1}$')
axes.set_xlim([34940.0, 35654.0])
axes.set_ylim([-383.1, 45946.0])
plt.show()
```
(3) Open the fit browser and add a `Bk2BkExpConvPV` peak - note that the values of `Gamma` and `Sigma2` will be overwritten as the FWHM of the peak is estimated. Check these values `A =0.1436 `and `B=0.0207` (may not match exactly depends on the exact center of the peak)



Fixes #31691 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
